free-4.md

```diff
@@ -0,0 +1,92 @@
+
+
+### Related work
+_Are there any other DSLs in this domain? If not, describe how you know there
+aren't and conjecture why not. If so, describe them and provide links. How well 
+do they address the need? Are there any particularly admirable qualities of the
+language? Are there parts of the language you think could be improved?_
+
+As previously mentioned, there are many specific tools available that focus on a single encoding. [Morse Code Decoder](http://morsecode.scphillips.com/translator.html) is one such example. It allows the user to translate between Morse Code and English. A different [encoder/decoder tool](http://www.splitbrain.org/services/encode) lets the user input text and lists how it appears in a lot of different encodings. The first decoder is limited in that it only converts between English and Morse Code, though it has a helpful GUI. The second decoder does not allow the user to input a message in anything besides English (Latin) or UTF-8. However, it shows the user many different ways of encoding the text.
+
+
+## The Project
+This section examines whether the idea makes for a good CS 111 project.
+
+
+### Suitability
+_If someone were to work on this project, what percentage of their time would be
+spent directly engaging in the **language** aspects of this project (e.g.,
+making language design decisions), as opposed to "systems" aspects of the
+project (e.g., implementing a complicated semantics that doesn't require a lot
+of language design)?_
+
+I think the majority of the project would be spend on the language aspects of the problem. Especially if the DSL were to use the CLI, little time would need to be spend on a UI. The challenge lies in deciding which encodings to include and how to handle the errors associated with translating between the various encodings. Tied to this problem are the challenges of storing each encoding and their various character mapping as well as deciding whether the user should be able to translate between characters beyond A-Z.
+
+
+### Scope
+_How big an idea is this? How ambitious is this project?_
+
+This is a fairly ambitious project. It seems fairly simple to explain on the surface but contains a lot of hidden problems that could easily cause a lot of trouble. It would be easy to start small and gradually expand the scope of the project.
+
+
+### Benefits and drawbacks
+_Why might this be a good idea for a project? Why might this not be a good idea 
+project?_
+
+I think it would be a good project to get experience focusing on a specific domain. It would be possible to make the language extensible and the project would involve a lot of challenging questions. It would not be a good project idea if the questions were difficult to solve or if it tried to do too many things with little organization. The risk with this idea is starting small and expanding the idea in inefficient directions.
+
```

Comment:
Just as a quick note, it kind of seems like a lot of these ideas are quite ambitious. If you are considering to do any of them, perhaps rescope the project idea to focus on only one particular aspect and maybe consider more the language design aspects over the systems aspect? I think they are all really cool ideas, nonetheless.